sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/BufferedHttpResponse.java

```diff
@@ -31,7 +31,8 @@ public BufferedHttpResponse(HttpResponse innerHttpResponse) {
         this.cachedBody = FluxUtil.collectBytesInByteBufferStream(innerHttpResponse.getBody())
             .map(ByteBuffer::wrap)
             .flux()
-            .cache();
+            .cache()
+            .map(ByteBuffer::duplicate);
```

Comment:
Why do we need to duplicate? The underlying byte array is still shared.

Reply:
Given the intent of this class is to be a buffered HttpResponse if we don't duplicate the ByteBuffers returned then it will only be able to be read once as the limit in the ByteBuffer will be equal to the count. Duplicating the ByteBuffer simply creates another view over the underlying byte array allowing for this to be consumed multiple times without concern.